tools/lint/src/main/kotlin/FirebaseAppGetDetector.kt

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.lint.checks
+
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.util.InheritanceUtil
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.UMethod
+import org.jetbrains.uast.getParentOfType
+
+@Suppress("DetectorIsMissingAnnotations")
+class FirebaseAppGetDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableMethodNames(): List<String> = listOf("get")
+
+  override fun visitMethodCall(context: JavaContext, call: UCallExpression, method: PsiMethod) {
+    if (!isFirebaseAppGet(method)) {
+      return
+    }
+
+    if (withinGetInstance(call)) {
+      return
+    }
+    call.getParentOfType<UMethod>() ?: return
```

Comment:
if we only want to check if it exists, IMO it would be easier to read inside an if statement